hummingbot/strategy/cross_exchange_market_making/cross_exchange_market_making.pyx

```diff
@@ -1023,51 +926,89 @@ cdef class CrossExchangeMarketMakingStrategy(StrategyBase):
             return self._suggested_price_samples[market_pair]
         return deque(), deque()
 
-    cdef c_take_suggested_price_sample(self, object market_pair, list active_orders):
+    cdef tuple c_get_top_bid_ask(self, object market_pair):
+        """
+        Calculate the top bid and ask using top depth tolerance in maker order book
+
+        :param market_pair: cross exchange market pair
+        :return: (top bid, top ask)
+        """
+        cdef:
+            OrderBook maker_order_book = market_pair.maker.order_book
+
+        if self._top_depth_tolerance == 0:
+
+            top_bid_price = maker_order_book.c_get_price(False)
+
+            top_ask_price = maker_order_book.c_get_price(True)
+
+        else:
+            # Use bid entries in maker order book
+            top_bid_price = maker_order_book.c_get_vwap_for_volume(False, self._top_depth_tolerance)
```

Comment:
The top depth tolerance setting means the order book depth to ignore, not average out. So getting the vwap here is wrong. It should be `c_get_price_for_volume()` instead for this.